### PR TITLE
upgrade rust to latest version

### PR DESCRIPTION
**Pull request by OmarTawfik**

### CODE REVIEW

File: crates/metaslang/cst/src/cursor.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ impl<T: KindTypes> Cursor<T> {
     }
 
     fn set_from_ancestor_node(&mut self, ancestor: &Rc<PathAncestor<T>>) {
-        self.parent = ancestor.parent.clone();
+        self.parent.clone_from(&ancestor.parent);
```
Comment (Xanewok) [inline crates/metaslang/cst/src/cursor.rs:52]: It's so rarely used in practice to the point where this looks less immediately clear than `a = b.clone()` for me but I don't want to block as it's a minor thing.

**Review by Xanewok — APPROVED** [review]
LGTM, thanks!

**Comment by Xanewok** [comment]
Removed from the queue in case you agree with https://github.com/NomicFoundation/slang/pull/1015#discussion_r1643020527, if not, feel free to re-add it again.

File: crates/metaslang/cst/src/cursor.rs (hunk 1 above)
Comment (OmarTawfik) [reply crates/metaslang/cst/src/cursor.rs:52]: @Xanewok this was produced by https://rust-lang.github.io/rust-clippy/master/index.html#/assigning_clones

I'm not sure how common it is to use `clone_from()` in external crates is, and I agree the code is a bit less readable this way. I don't see a way to silence this rule on specific types (`Rc<>`, etc...). Should we just disable it?

Comment (Xanewok) [reply crates/metaslang/cst/src/cursor.rs:52]: I think so. We can re-enable it once we want to eek out more performance (the `Clone::clone_from` exists to reuse certain allocations when cloning).

(fun fact, the `derive(Clone)` machinery does not use that and uses `Clone::clone` directly due to the sheer codegen impact and thus boostrap/compile timings in rustc while not always bringing perf benefits).

**Pull request merged**